app/graphql/types/concerns/profile_scoring.rb

```diff
@@ -5,12 +5,22 @@ module Types
   module ProfileScoring
     extend ActiveSupport::Concern
 
+    def unsupported_host_count
+      ::CollectionLoader.for(
+        object.class,
+        test_result_method
+      ).load(object).then do |test_results|
+        test_results.latest.supported(false).count
+      end
+    end
+
     def compliant_host_count
       ::CollectionLoader.for(
         object.class,
-        object.policy_id ? :policy_test_result_hosts : :test_result_hosts
-      ).load(object).then do |hosts|
-        hosts.count { |host| object.compliant?(host) }
+        test_result_method
+      ).load(object).then do |test_results|
+        Host.where(id: test_results.latest.supported.select(:host_id))
```

Comment:
Could it not be `test_results.hosts...` or is that not possible due to GraphQL?

Reply:
We're joining multiple test results and multiple hosts. What you're thinking of maybe is `test_result.hosts`; it'd be like `test_results.map(&:hosts)` here, which would create N queries.